rtfs.go

```diff
@@ -19,15 +19,24 @@ type IpfsManager struct {
 }
 
 // NewManager is used to initialize our Ipfs manager struct
-func NewManager(ipfsURL string, timeout time.Duration, direct bool) (*IpfsManager, error) {
-	// instantiate shell
-	sh := newShell(ipfsURL, direct)
-	// set timeout
-	sh.SetTimeout(timeout)
+func NewManager(ipfsURL, token string, timeout time.Duration) (*IpfsManager, error) {
+	var (
+		sh  *ipfsapi.Shell
+		err error
+	)
+	if token != "" {
+		sh = ipfsapi.NewDirectShell(ipfsURL).WithAuthorization(token)
+		_, err = sh.ID()
+	} else {
+		sh = ipfsapi.NewShell(ipfsURL)
```

Comment:
might want to note in the `NewManager` documentation that we create a direct shell (ie skipping multiaddr parsing) if and only if a token is provided

Reply:
Good point